src/DAG/dag.c

```diff
@@ -573,10 +536,98 @@ void RedisAI_BatchedDagRunSessionStep(RedisAI_RunInfo **batched_rinfo, const cha
             __atomic_store_n(rinfo->dagError, 1, __ATOMIC_RELAXED);
         }
     }
-
     return;
 }
 
+static int StoreTensorInKeySpace(RedisModuleCtx *ctx, RAI_Tensor *tensor,
+                                 const char *persist_key_name, bool mangled_name) {
+    int ret = REDISMODULE_ERR;
+    RedisModuleKey *key;
+    char *demangled_key_name = RedisModule_Strdup(persist_key_name);
+    if (mangled_name)
+        demangled_key_name[strlen(persist_key_name) - 4] = 0;
+    RedisModuleString *tensor_keyname =
+        RedisModule_CreateString(ctx, demangled_key_name, strlen(demangled_key_name));
+    const int status =
+        RAI_OpenKey_Tensor(ctx, tensor_keyname, &key, REDISMODULE_READ | REDISMODULE_WRITE);
+    RedisModule_Free(demangled_key_name);
+    if (status == REDISMODULE_ERR) {
+        RedisModule_ReplyWithError(ctx, "ERR could not save tensor");
+        goto clean_up;
+    } else {
+        if (RedisModule_ModuleTypeSetValue(key, RedisAI_TensorType,
+                                           RAI_TensorGetShallowCopy(tensor)) != REDISMODULE_OK) {
+            RedisModule_ReplyWithError(ctx, "ERR could not save tensor");
+            goto clean_up;
+        }
+    }
+    ret = REDISMODULE_OK;
+clean_up:
+    RedisModule_CloseKey(key);
+    RedisAI_ReplicateTensorSet(ctx, tensor_keyname, tensor);
+    return ret;
+}
+
+static void PersistTensors(RedisModuleCtx *ctx, RedisAI_RunInfo *rinfo) {
+    AI_dictIterator *persist_iter = AI_dictGetSafeIterator(rinfo->dagTensorsPersistedContext);
+    AI_dictEntry *persist_entry = AI_dictNext(persist_iter);
+    while (persist_entry) {
+        const char *persist_key_name = AI_dictGetKey(persist_entry);
+
+        AI_dictEntry *tensor_entry = AI_dictFind(rinfo->dagTensorsContext, persist_key_name);
+
+        if (tensor_entry) {
+            RAI_Tensor *tensor = AI_dictGetVal(tensor_entry);
+
+            if (tensor == NULL) {
+                persist_entry = AI_dictNext(persist_iter);
+                continue;
```

Comment:
In which cases such a situation could happen and why we would like to continue without error?

Reply:
There is a test (test_onnx_modelrun_iris) where more than one output tensor was specified in the modelRunCtx, but the model returns only one tensor as output. In this case the second tensor is NULL.